### PR TITLE
Bring INFINITY constants up to current guidelines

### PR DESCRIPTION
**Pull request by kgaillot**

### CODE REVIEW

**Comment by kgaillot** [comment]
@nrwahl2 , here's another best-practices PR. Low priority

File: include/crm/common/util.h [hunk 1]
```diff
@@ -41,8 +42,6 @@ char *pcmk_promotion_score_name(const char *rsc_id);
 int crm_default_remote_port(void);
 
 /* public score-related functions (from scores.c) */
-const char *pcmk_readable_score(int score);
-int char2score(const char *score);
 int pcmk__add_scores(int score1, int score2);
```
Comment (nrwahl2) [inline include/crm/common/util.h:45]: Drop this and the "public score-related functions" comment

File: lib/common/xml.c [hunk 2]
```diff
@@ -563,8 +563,8 @@ expand_plus_plus(xmlNode * target, const char *name, const char *value)
     }
     int_value += offset;
 
-    if (int_value > INFINITY) {
-        int_value = (int)INFINITY;
```
Comment (nrwahl2) [inline lib/common/xml.c:567]: That cast was odd

**Review by nrwahl2 — COMMENTED** [review]
Nice

File: lib/common/tests/scores/char2score_test.c [hunk 3]
```diff
@@ -56,8 +56,10 @@ special_values(void **state)
 static void
 outside_limits(void **state)
 {
-    assert_int_equal(char2score(B(CRM_SCORE_INFINITY) "00"), CRM_SCORE_INFINITY);
-    assert_int_equal(char2score("-" B(CRM_SCORE_INFINITY) "00"), -CRM_SCORE_INFINITY);
+    assert_int_equal(char2score(B(CRM_SCORE_INFINITY) "00"),
```
Comment (nrwahl2) [inline lib/common/tests/scores/char2score_test.c:59]: These `CRM_SCORE_INFINITY` occurrences in `char2score()` didn't get updated to `PCMK_SCORE_INFINITY`, which is probably what's causing the unit test to fail.

Comment (kgaillot) [reply lib/common/tests/scores/char2score_test.c:59]: whoops, thanks

**Comment by kgaillot** [comment]
Updated per review

**Review by nrwahl2 — APPROVED** [review]

**Pull request merged**